From 0dc4b541bcaf52b1110b14968e8c6986901d041e Mon Sep 17 00:00:00 2001
From: Beniamino Galvani <bgalvani@redhat.com>
Date: Tue, 27 Oct 2015 20:37:30 +0100
Subject: [PATCH] core: ignore slave devices when deciding global state

The state of slave devices is not meaningful for deciding the global
state as the IP configuration is done on master device: ignore them in
nm_manager_update_state().

https://bugzilla.redhat.com/show_bug.cgi?id=951399
---
 src/nm-manager.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/nm-manager.c b/src/nm-manager.c
index c6d4562..701b6ee 100644
--- a/src/nm-manager.c
+++ b/src/nm-manager.c
@@ -457,7 +457,13 @@ nm_manager_update_state (NMManager *manager)
 
 		for (iter = priv->devices; iter; iter = iter->next) {
 			NMDevice *dev = NM_DEVICE (iter->data);
+			NMActRequest *request;
 
+			/* Skip slave devices */
+			request = nm_device_get_act_request (dev);
+			if (request && nm_act_request_get_master (request))
+				continue;
+				
 			if (nm_device_get_state (dev) == NM_DEVICE_STATE_ACTIVATED) {
 				new_state = NM_STATE_CONNECTED;
 				break;
-- 
2.4.3

